PEPit/functions/smooth_convex_function.py

```diff
@@ -43,6 +44,11 @@ def __init__(self,
         # Inherit from SmoothStronglyConvexFunction as a special case of it with mu=0.
         super().__init__(is_leaf=is_leaf,
                          decomposition_dict=decomposition_dict,
-                         reuse_gradient=True,
+                         reuse_gradient=reuse_gradient,
                          mu=0,
                          L=L)
+
+        if self.L == np.inf:
+            print("\033[96m(PEPit) The class of smooth convex functions is necessarily differentiable.\n"
+                  "When setting L to infinity, please use the class of convex functions (ConvexFunction),\n "
+                  "that allows to compute several subgradients at the same point each time one is required.\033[0m")
```

Comment:
The class of XXXX functions is necessarily differentiable. 
When setting L to infinity, please use the class of XXXX functions (XXX),
(which accounts for the fact there might be several subgradients at the same point).